planner/core/preprocess_test.go

```diff
@@ -255,6 +255,11 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"CREATE TABLE origin (a int primary key auto_increment, b int);", false, nil},
 		{"CREATE TABLE origin (a int unique auto_increment, b int);", false, nil},
 		{"CREATE TABLE origin (a int key auto_increment, b int);", false, nil},
+		// issue 18149
+		{"CREATE TABLE t (a int, index ``(a));", true, errors.New("[ddl:1280]Incorrect index name ''")},
+		{"CREATE TABLE t (a int, key ``(a));", true, errors.New("[ddl:1280]Incorrect index name ''")},
+		{"CREATE TABLE t (a int, index(a));", false, nil},
+		{"CREATE TABLE t (a int,key(a));", false, nil},
```

Comment:
Please add the following tests and make sure `Incorrect index name ''` is raised:

* Create expression index in `CREATE TABLE`:
```
create table t(a varchar(10), index `` ((lower(a))));
```

* Create column index by `CREATE INDEX`:
```
create index `` on t (a);
```
* Create expression index by `CREATE INDEX`:
```
create index `` on t ((lower(a)));
```

Reply:
the lenght of  Index name  must greater than 0  in CreateIndexStmt.